plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/connection/Connection.java

```diff
@@ -483,6 +484,23 @@ public boolean equals(Object obj) {
 		return true;
 	}
 
+	@Override
+	public boolean credentialsEqual(IConnection connection) {
+		if(!equals(connection)) {
+			return false;
+		}
+		//It is safe to cast now.
+		Connection other = (Connection)connection;
+		//User name is already compared
+		if(!Objects.equals(password, other.password)) {
```

Comment:
I assume that a purely token based connection will always have null in the password? 
I admit that this part is rather fuzzy in our current Connection implementation, we have the authentication strategies but then we still have the password and username in connection. We should clean this stuff up at some point I think